felix/nftables/ipsets.go

```diff
@@ -404,47 +405,46 @@ func (s *IPSets) tryResync() error {
 	type setData struct {
 		setName string
 		elems   []*knftables.Element
-		err     error
 	}
-	setsChan := make(chan setData)
-	defer close(setsChan)
 
 	// Start a goroutine to list the elements of each set. Limit concurrent set reads to
 	// avoid spawning too many goroutines if there are a large number of sets.
 	routineLimit := make(chan struct{}, 100)
 	defer close(routineLimit)
-	for _, setName := range sets {
+
+	// Create an errgroup to wait for all the set reads to complete.
+	g, ctx := errgroup.WithContext(ctx)
+	responses := make([]setData, len(sets))
+
+	for i, name := range sets {
 		// Wait for room in the limiting channel.
 		routineLimit <- struct{}{}
```

Comment:
errgroup has a built-in limiter, just need to use `g.SetLimit()` and then you can get rid of the custom channel